src/Versions.jl

```diff
@@ -89,7 +89,7 @@ function isjoinable(up::VersionBound, lo::VersionBound)
     return true
 end
 
-Base.hash(r::VersionBound, h::UInt) = hash(hash(r.t, h), r.n)
+Base.hash(r::VersionBound, h::UInt) = hash(hash(r.t, h), convert(UInt64, r.n))
```

Comment:
Why convert to `UInt64`?

Reply:
Just to avoid this. There may be a more correct solution
```
julia> hash(Pkg.Types.VersionBound())
ERROR: MethodError: no method matching hash(::UInt64, ::Int64)
Closest candidates are:
  hash(::UInt64, ::UInt64) at hashing.jl:72
  hash(::Real, ::UInt64) at float.jl:489
  hash(::Any) at hashing.jl:18
  ...
Stacktrace:
 [1] hash(r::Pkg.Types.VersionBound, h::UInt64)
   @ Pkg.Types /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Pkg/src/versions.jl:88
 [2] hash(x::Pkg.Types.VersionBound)
   @ Base ./hashing.jl:18
 [3] top-level scope
   @ REPL[2]:1
```

Reply:
Ah, I think the correct way is to write it as:

```jl
hash(r.t, hash(r.n, h))
```